src/coreclr/src/jit/morph.cpp

```diff
@@ -3857,7 +3857,7 @@ GenTreeCall* Compiler::fgMorphArgs(GenTreeCall* call)
                                 copyBlkClass = objClass;
                             }
                         }
-                        else if (genActualType(varDsc->TypeGet()) != structBaseType)
+                        else if (genTypeSize(varDsc->TypeGet()) != genTypeSize(structBaseType))
```

Comment:
It may be that it's not possible, though I wouldn't bet on it. But if the sizes are the same we can safely retype as long as we handle any register file mismatches in the backend. This is simply loosening this condition.